src/jit/gtlist.h

```diff
@@ -161,7 +161,7 @@ GTNODE(ADD_LO     , "+Lo"          ,1,GTK_BINOP)
 GTNODE(ADD_HI     , "+Hi"          ,1,GTK_BINOP)
 GTNODE(SUB_LO     , "-Lo"          ,0,GTK_BINOP)
 GTNODE(SUB_HI     , "-Hi"          ,0,GTK_BINOP)
-GTNODE(MUL_HI     , "*Hi"          ,1,GTK_BINOP)
+GTNODE(MUL_HI     , "*Hi"          ,1,GTK_BINOP) // A mul that returns the 2N bit result of an NxN multiply
```

Comment:
It would seem so. The problem, as I see it, is that the existing `GT_MULHI` is basically the 64/128-bit version of these `GT_XXX_HI` ops that are (AFAICT) used only for 32/64. Also, it seems like we would want to add a comment that, at least for now, we don't have a 32-bit target that requires the `GT_XXX_HI` equivalent, as the full 64 bits can be computed with a single op. It may be the case, in future, that we will have a target that requires it (I'm not sure what arm32 does for long multiply). And I would add a comment to `GT_MULHI` to clarify that it is for the 64/128-bit case, as contrasted with the `GT_XXX_HI` ops that are for the 32/64-bit case.

---
In reply to: [77701031](https://github.com/dotnet/coreclr/pull/7065#discussion_r77701031) [](ancestors = 77701031)